### PR TITLE
 Adding a singlenode setup

### PR DESCRIPTION
**Pull request by robert-zaremba**
For smart contract development we don't need a whole net. It's enough to have only a single node setup.

### CODE REVIEW

**Comment by robert-zaremba** [comment]
Sorry for spamming with the commits. I was testing & fixing Circle configuration and finally it worked.

FYI: I removed the deprecated check (`/internal`)

File: .circleci/config.yml [hunk 1]
```diff
@@ -25,13 +29,6 @@ commands:
             INTERNAL_ADDR=localhost:<< parameters.port >> \
             WEBSOCKET_ADDR=localhost:<< parameters.port >> \
             ./test/healthcheck.sh
-      - run:
```
Comment (dincho) [inline .circleci/config.yml]: Could you please keep this, I plan to include this new config into the 1.0 (legacy) tag?

File: test/healthcheck.sh [hunk 2]
```diff
@@ -7,19 +7,19 @@ INTERNAL_ADDR=${INTERNAL_ADDR:-localhost:3113}
...
+echo "Testing external API"
+curl -sSf -o /dev/null --retry 6 http://${EXTERNAL_ADDR}/v2/status || exit 2
 
-# Internal API
+echo "Testing internal API"
 PEERS_COUNT=$(curl -sS ${INTERNAL_ADDR}/v2/debug/peers | grep -o aenode | wc -l)
 
-# Explicit exit because otherwise test would exit with status 127
-test $PEERS_COUNT -ge $MIN_PEERS || exit 1
+echo "Testing internal API - amount of peers="$PEERS_COUNT
```
Comment (dincho) [inline test/healthcheck.sh]: What about `echo "Testing internal API - $PEERS_COUNT peers` ?

File: test/healthcheck.sh [hunk 3]
```diff
@@ -7,19 +7,19 @@ INTERNAL_ADDR=${INTERNAL_ADDR:-localhost:3113}
 WEBSOCKET_ADDR=${WEBSOCKET_ADDR:-localhost:3014}
```
Comment (dincho) [inline test/healthcheck.sh]: Nice improvements to this script 👍

File: .circleci/config.yml (hunk 1 above)
Comment (robert-zaremba) [reply .circleci/config.yml]: done

File: test/healthcheck.sh (hunk 2 above)
Comment (robert-zaremba) [reply test/healthcheck.sh]: good idea. done.

**Comment by robert-zaremba** [comment]
Hi, I've followed your recommendation and sqashed the commits.

**Comment by dincho** [comment]
Thank you @robert-zaremba

**Pull request merged**